src/pages/resources.js

```diff
@@ -32,55 +32,87 @@ class Resources extends Component {
 		status: "Loading..."
 	};
 
-	searchResources = event => {
-		const query = event.target.value.toLowerCase();
+	filterResources = async () => {
+		const resourcesAndFilters = this.getResourcesAndFilters();
 
-		if (query === "") {
-			this.setState({ resources: this.state.constantResources });
-		} else {
-			const resourcesThatMatchQuery = [];
+		resourcesAndFilters[0] = this.getFilteredResources(...resourcesAndFilters);
+		this.changeUrl(resourcesAndFilters[2]);
 
-			for (const resource of this.state.constantResources) {
-				if (resource.name.toLowerCase().includes(query)) {
-					resourcesThatMatchQuery.push(resource);
-				}
-			}
+		this.setState({
+			resources: resourcesAndFilters[0],
+			filterResources: resourcesAndFilters[1],
+			filterPrice: resourcesAndFilters[2]
+		});
 
-			if (!resourcesThatMatchQuery.length) {
-				this.setState({status: "No matches found..."});
-			}
+		const noMatchesFound = resourcesAndFilters[0].length !== 0;
 
-			this.setState({
-				status: null,
-				resources: resourcesThatMatchQuery
-			});
+		if (noMatchesFound) {
+			this.setState({ status: "No matches found..." });
 		}
 	};
 
-	filterResources = async event => {
-		let resources = this.state.constantResources;
-		const value = event.target.value;
-
-		if (value !== "Show All") {
-			resources = await resources.filter(resource => resource.category.toLowerCase() === value);
-			this.props.router.push(`/resources?category=${value}`, undefined, { shallow: true });
-		} else {
-			this.props.router.push("/resources", undefined, { shallow: true });
-		}
+	getResourcesAndFilters() {
+		const resources = this.state.constantResources;
+
+		const categoryFilter = document.getElementsByClassName(
+			"resources__filter-category"
+		)[0].value;
+		const priceFilter = document.getElementsByClassName(
+			"resources__filter-price"
+		)[0].value;
+		const searchFilter = document.getElementsByClassName(
+			"resources__search-filter"
+		)[0].value;
+
+		const resourcesAndFilters = [
+			resources,
+			categoryFilter,
+			priceFilter,
+			searchFilter
+		];
+
+		return resourcesAndFilters;
```

Comment:
I would just return the array here rather than assign it to a variable and returning that.